docs/literate/src/files/behind_the_scenes_simulation_setup.jl

```diff
@@ -0,0 +1,246 @@
+#   This is possible because, from a storage perspective, they share the same stored data, while
+#   access to this data is provided in different ways.
+
+
+# - `compute_coefficients!(u, initial_conditions, t, mesh::DG, equations, solver, cache)`
+
+#   Now the variable `u`, intended to store solutions, has been allocated and wrapped, it is time
+#   to apply the initial conditions. The `compute_coefficients!` function calculates the initial
+#   conditions for each variable at every node within each element and properly stores them in the
+#   `u` array.
+
+# At this stage, the `semidiscretize` function has all the necessary components to initialize and
+# return an `ODEProblem` object, which will be used by the `solve` function to compute the
+# solution.
+
+# In summary, the internal workings of `semidiscretize` with brief descriptions can be presented
+# as follows.
+
+# ![semidiscretize_structure](https://github.com/trixi-framework/Trixi.jl/assets/119304909/491eddc4-aadb-4e29-8c76-a7c821d0674e)
+
+# ## Functions `solve` and `rhs!`
+
+# Once the `ODEProblem` object is initialized, the `solve` function and one of the ODE solvers from
+# the OrdinaryDiffEq.jl package can be utilized to compute an approximated solution using the
+# instructions contained in the `ODEProblem` object.
+
+sol = solve(ode, CarpenterKennedy2N54(williamson_condition = false), dt = 0.01,
+            save_everystep = false);
+
+# Since the `solve` function and the ODE solver have no knowledge
+# of a particular spatial discretization, it is necessary to define a
+# "right-hand-side function", `rhs!`, within Trixi.jl.
+
+# Trixi.jl includes a set of `rhs!` functions designed to compute `du`, i.e.,
+# ``\partial u/\partial t`` according to the structure
+# of the setup. These `rhs!` functions calculate interface, mortars, and boundary fluxes, in
+# addition to surface and volume integrals, in order to construct the `du` vector. This `du` vector
+# is then used by the time integration method to obtain the solution at the subsequent time step.
+# The `rhs!` function is called by time integration methods in each iteration of the solve loop
+# within OrdinaryDiffEq.jl, with arguments `du`, `u`, `semidiscretization`, and the current time.
+
+# The problem is that `rhs!` functions within Trixi.jl are specialized for specific solver and mesh
+# types. However, the types of arguments passed to `rhs!` by time integration methods do not
+# explicitly provide this information. Consequently, Trixi.jl uses a two-levels approach for `rhs!`
+# functions. The first level is limited to a single function for each `semidiscretization` type,
+# and its role is to redirect data to the target `rhs!`. It performs this by extracting the
+# necessary data from the integrator and passing them, along with the originally received
+# arguments, to the specialized for solver and mesh types `rhs!` function, which is
+# responsible for calculating `du`.
```

Comment:
I think in this version this causes more confusion than enlightment - maybe this becomes more clear if you just elaborate why there is a `rhs!` for the semidiscretizations and then go to the specific solvers.

Reply:
You are right!

Is it now better:

# Trixi.jl uses a two-levels approach for `rhs!` functions. The first level is limited to a
# single function for each `semidiscretization` type, and its role is to redirect data to the
# target `rhs!` for specific solver and mesh types. This target `rhs!` function is responsible
# for calculating `du`.